src/slash/open.ts

```diff
@@ -166,18 +182,22 @@ export class RegisterButtonHandler implements ButtonClickHandler {
 			await interaction.user.send("Sorry, the tournament is currently full!");
 			return;
 		}
-		const modal = new ModalBuilder().setCustomId("registerModal").setTitle(`Register for ${tournament.name}`);
-		const deckLabelInput = new TextInputBuilder()
-			.setCustomId("friendCode")
-			.setLabel("Master Duel Friend Code")
-			.setStyle(TextInputStyle.Short)
-			.setRequired(tournament.requireFriendCode)
-			.setPlaceholder("000000000")
-			.setMinLength(9)
-			.setMaxLength(11);
-		const actionRow = new ActionRowBuilder<TextInputBuilder>().addComponents(deckLabelInput);
-		modal.addComponents(actionRow);
-		await interaction.showModal(modal);
+		if (tournament.requireFriendCode) {
+			const modal = new ModalBuilder().setCustomId("registerModal").setTitle(`Register for ${tournament.name}`);
+			const deckLabelInput = new TextInputBuilder()
+				.setCustomId("friendCode")
+				.setLabel("Master Duel Friend Code")
+				.setStyle(TextInputStyle.Short)
+				.setRequired(tournament.requireFriendCode)
```

Comment:
If we're only asking for the friend code when required (which is a fair direction to take it), this should just be `true`